Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -284,6 +292,37 @@ public final class SwiftTargetBuildDescription {
                 self.resourceBundleInfoPlistPath = infoPlistPath
             }
         }
+
+        try self.generateResourceEmbeddingCode()
+    }
+
+    // FIXME: This will not work well for large files, as we will store the entire contents, plus its byte array representation in memory and also `writeIfChanged()` will read the entire generated file again.
+    private func generateResourceEmbeddingCode() throws {
+        guard needsResourceEmbedding else { return }
+
+        let stream = BufferedOutputByteStream()
+        stream <<< """
+        struct PackageResources {
+
+        """
+
+        try resources.forEach {
+            guard $0.rule == .embedInCode else { return }
+
+            let variableName = $0.path.basename.spm_mangledToC99ExtendedIdentifier()
```

Comment:
could we run into collisions?

Reply:
Yah, since there can be subdirectories and this will become a flat namespace. I guess we could add the relative path to the target directory, either in general or for collisions.

Reply:
Looks like this case is actually already handled at the `TargetSourcesBuilder` level and not allowed. So I think we won't handle it here, either.